### PR TITLE
New footer design

### LINKED ISSUE #478: Improve the Footer Design
### **Description:**
The current footer design lacks visual appeal and does not meet modern design standards. The goal of this issue is to enhance the footer by:

- Adding Icons: Incorporate icons for GitHub, YouTube, LinkedIn, and Contact Us.
- Modular Design: Redesign the footer to be modular, ensuring it adheres to industry standards for responsiveness and usability.

**Suggested Design**

Screenshot 2024-08-05 175723

### PR DESCRIPTION
**Pull request by ayush2743**
FIXED #478 

Improved the footer Design

### CODE REVIEW

File: components/Footer.tsx [hunk 1]
```diff
@@ -1,70 +1,94 @@
 import Image from "next/image";
 import { env } from "@/env.mjs";
+import { FaYoutube, FaLinkedin, FaGithub, FaEnvelope } from "react-icons/fa";
+
+const SocialLink = ({ href, icon: Icon, label }) => (
+  <a
```
Comment (rithviknishad) [inline components/Footer.tsx:6]: Let's use next/link's Link component instead

File: components/Footer.tsx [hunk 2]
```diff
@@ -1,70 +1,94 @@
...
+    href={href}
```
Comment (rithviknishad) [inline components/Footer.tsx:7]: Let's open links in new tab and also add appropriate `rel` values

```suggestion
    href={href}
    target="_blank"
    rel="noopener noreferrer"
```

**Review by rithviknishad — COMMENTED** [review]
LGTM, just few minor things...

**Comment by ayush2743** [comment]
Is there a problem in merging this PR?
As i can see, some checks were unsuccessful.

**Comment by rithviknishad** [comment]
@ayush2743 Need types as this is a typescript project. Also needs above review changes requested to be made.

image

**Comment by ayush2743** [comment]
@rithviknishad Can you review it once again please?

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**